### DISCUSSION TITLE
Linting and formatting

### DISCUSSION THREAD

**Discussion by DhivinX** [xeokit/sdk#24] [General]
Hello!
I'm working on eslint and prettier setup and I have a few questions.

1. Do we have any specific requirements regarding linting?
2. What specific rules should be disabled?
3. For formatting, can we indent 2 spaces or 4 spaces? I will add that the current standards are 2 spaces, but I will set it as you like.
4. Things that cannot be fixed automatically by eslint, e.g. using var instead of const/let, should I leave it or try to fix it myself?

Rules what I suggested to off for this repo:

```
'@typescript-eslint/no-floating-promises': 'off',
'@typescript-eslint/require-await': 'off',
'@typescript-eslint/no-explicit-any': 'off',
'@typescript-eslint/no-unsafe-call': 'off',
'@typescript-eslint/no-unsafe-member-acces': 'off',
'@typescript-eslint/no-empty-interface': 'off',
'@typescript-eslint/ban-ts-comment': 'off',
'@typescript-eslint/no-unsafe-declaration-merging': 'off',
'@typescript-eslint/no-unused-vars': ['warn', { argsIgnorePattern: '^_' }],
```

**Comment by xeolabs** [comment]
Hi,

> 1. Do we have any specific requirements regarding linting?
> 2. What specific rules should be disabled?

In my IDE (WebSTorm)  actually had eslint disabled, and tslint disabled as well, so nothing comes to mind - my IDE, WebStorm is running a TypeScript inspector of its ow, but I ca't determine where the rules are coming from.

> 3. For formatting, can we indent 2 spaces or 4 spaces? I will add that the current standards are 2 spaces, but I will set it as you like.

Let's make it four spaces. 

> 4. Things that cannot be fixed automatically by eslint, e.g. using var instead of const/let, should I leave it or try to fix it myself?

Just leave any violations you find - I'll go through and fix up any lint errors.

> Rules what I suggested to off for this repo:
> 
> ```
> '@typescript-eslint/no-floating-promises': 'off',
> '@typescript-eslint/require-await': 'off',
> '@typescript-eslint/no-explicit-any': 'off',
> '@typescript-eslint/no-unsafe-call': 'off',
> '@typescript-eslint/no-unsafe-member-acces': 'off',
> '@typescript-eslint/no-empty-interface': 'off',
> '@typescript-eslint/ban-ts-comment': 'off',
> '@typescript-eslint/no-unsafe-declaration-merging': 'off',
> '@typescript-eslint/no-unused-vars': ['warn', { argsIgnorePattern: '^_' }],
> ```

Yip those look good to me.

If it's practical, perhaps we can disable linting for now?

**Reply by DhivinX** [reply]
Everything is clear, thanks for the answers.  Regarding your question, you can add a comment in any file, add .eslitignore or disable rules from the preset to disable linting.  After testing I can disable eslint if you want.

**Reply by xeolabs** [reply]
Yeah let's disable linting for now.
